nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -102,8 +107,26 @@
             .defaultValue(WRITE_CONCERN_ACKNOWLEDGED)
             .build();
 
+    static final PropertyDescriptor RESULTS_PER_FLOWFILE = new PropertyDescriptor.Builder()
+            .name("results-per-flowfile")
+            .displayName("Results Per FlowFile")
+            .description("How many results to put into a flowfile at once. The whole body will be treated as a JSON array of results.")
+            .required(false)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("1")
+            .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .description("The number of elements returned from the server in one batch")
+            .required(false)
```

Comment:
Do you want to add default value as 1, can get rid of null check in onTrigger method by doing this.